compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -65,15 +66,50 @@ impl<'a> ArchiveBuilder<'a> for LlvmArchiveBuilder<'a> {
     where
         F: FnMut(&str) -> bool + 'static,
     {
-        let archive_ro = match ArchiveRO::open(archive) {
+        let mut archive = archive.to_path_buf();
+        if self.sess.target.llvm_target.contains("-apple-macosx") {
+            let mut f = fs::File::open(&archive)?;
+            let mut prefix = [0; 4];
+            f.read(&mut prefix)?;
+            if prefix == *b"\xCA\xFE\xBA\xBE" {
+                let (_, extracted_path) = tempfile::Builder::new()
+                    .suffix(archive.as_path().file_name().unwrap())
+                    .tempfile()?
+                    .keep()
+                    .unwrap();
+                let result = std::process::Command::new("lipo")
```

Comment:
I believe the object crate supports reading fat libraries too. Once https://github.com/rust-lang/rust/pull/97485 lands that would allow to avoid writing a temporary file.

Reply:
Fantastic. That's way less horrifying :-) And it looks like it may in fact support these out of the box! Looking forward to that.

Reply:
I think some extra code will still be necessary for handling fat libraries. Am I right in that fat libraries use the same wrapper format that fat binaries do except that it wraps an archive instead of a Mach-O binary? There would still need to be some extra code to extract the right archive from the fat library using `object::macho::FatHeader::parse_arch32/64`.

Reply:
I know nothing about the on disk format (except the CAFEBABE prefix), but I suppose it makes sense that at least a little code will be required to extract sub-archive.